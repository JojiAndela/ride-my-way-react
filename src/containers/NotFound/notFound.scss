@import '../../assets/styles/_settings.scss';

.body{
  display:block;
  background-color:#2f3242;

.svg {
  position: absolute;
  top: 50%;
  left: 50%;
  margin-top: -250px;
  margin-left: -400px;
}
.messageBox {
  height: 200px;
  width: 380px;
  position: absolute;
  top: 50%;
  left: 50%;
  margin-top: -100px;
  margin-left: 50px;
  color: $gray;
  font-family: Roboto;
  font-weight: 300;
}
.messageBox h1 {
  font-size: 60px;
  line-height: 46px;
  margin-bottom: 40px;
}
.buttonsCon .actionLinkWrap {
  margin-top: 40px;
}
.buttonsCon .actionLinkWrap a {
  background: #68c950;
  padding: 8px 25px;
  border-radius: 4px;
  color: #FFF;
  font-weight: bold;
  font-size: 14px;
  transition: all 0.3s linear;
  cursor: pointer;
  text-decoration: none;
  margin-right: 10px
}
.buttonsCon .actionLinkWrap a:hover {
  background: #5A5C6C;
  color: #fff;
}

.Polygon1 , .Polygon2 , .Polygon3 , .Polygon4 , .Polygon4, .Polygon5 {
  animation: float 1s infinite ease-in-out alternate;
}
.Polygon2 {
  animation-delay: .2s;
}
.Polygon3 {
  animation-delay: .4s;
}
.Polygon4 {
  animation-delay: .6s;
}
.Polygon5 {
  animation-delay: .8s;
}

@keyframes float {
  100% {
    transform: translateY(20px);
  }
}
@media (max-width: 450px) {
  .svg {
    position: absolute;
    top: 50%;
    left: 50%;
    margin-top: -250px;
    margin-left: -190px;
  }
  .messageBox {
    top: 50%;
    left: 50%;
    margin-top: -100px;
    margin-left: -190px;
    text-align: center;
  }
}
}