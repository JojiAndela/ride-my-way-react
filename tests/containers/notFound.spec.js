import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
// Component to be tested
import NotFound from '../../src/containers/NotFound/NotFound';

let wrapper;
describe('NotFound', () => {
  beforeEach(() => {
    wrapper = mount(
        <BrowserRouter>
          <NotFound />
        </BrowserRouter>
    );
  });
  it('should render the not found page correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
