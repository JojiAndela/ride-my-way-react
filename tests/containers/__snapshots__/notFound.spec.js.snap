// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should render the not found page correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <notFound>
      <div
        className="body"
      >
        <svg
          className="svg"
          height="500px"
          version="1.1"
          viewBox="0 0 837 1045"
          width="380px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            className="Page1"
            fill="none"
            fillRule="evenodd"
            stroke="none"
            strokeWidth="1"
          >
            <path
              className="Polygon1"
              d="M353,9 L626.664028,170 L626.664028,487 L353,642 L79.3359724,487 L79.3359724,170 L353,9 Z"
              stroke="#007FB2"
              strokeWidth="6"
            />
            <path
              className="Polygon2"
              d="M78.5,529 L147,569.186414 L147,648.311216 L78.5,687 L10,648.311216 L10,569.186414 L78.5,529 Z"
              stroke="#EF4A5B"
              strokeWidth="6"
            />
            <path
              className="Polygon3"
              d="M773,186 L827,217.538705 L827,279.636651 L773,310 L719,279.636651 L719,217.538705 L773,186 Z"
              stroke="#795D9C"
              strokeWidth="6"
            />
            <path
              className="Polygon4"
              d="M639,529 L773,607.846761 L773,763.091627 L639,839 L505,763.091627 L505,607.846761 L639,529 Z"
              stroke="#F2773F"
              strokeWidth="6"
            />
            <path
              className="Polygon5"
              d="M281,801 L383,861.025276 L383,979.21169 L281,1037 L179,979.21169 L179,861.025276 L281,801 Z"
              stroke="#36B455"
              strokeWidth="6"
            />
          </g>
        </svg>
        <div
          className="messageBox"
        >
          <h1>
            404
          </h1>
          <p>
            Page not found
          </p>
          <div
            className="buttonsCon"
          >
            <div
              className="actionLinkWrap"
            >
              <Link
                className="linkButton"
                replace={false}
                to="/"
              >
                <a
                  className="linkButton"
                  href="/"
                  onClick={[Function]}
                >
                  Go to Home Page
                </a>
              </Link>
            </div>
          </div>
        </div>
      </div>
    </notFound>
  </Router>
</BrowserRouter>
`;
